### DISCUSSION TITLE
Tasmota web server is not accesible until running a network scan

### DISCUSSION THREAD

**Discussion by Android-X13** [arendst/Tasmota#16555] [Q&A]
I have a custom Termux task on my phone that sends specific web requests to an IR controller which is flashed with Tasmota. However 1 out of 5 times it fails to execute because Tasmota's web server apparently is not accesible for some reason.

If I repeat the task, it keeps failing, but I have noticed that I can successfully "wake up" the web server by running an nmap scan on the local network. Even the network scan usually doesn't find the device at the first try, so I have created another Termux task which runs the nmap scan in a loop until the Tasmota device responds. When I get a hit and the task exits successfuly, I can then continue sending my web requests normally until the next time.

Can anyone explain why this happens (as well as why nmap is able to "wake up" the device) and how it can be mitigated?

**Comment by s-hadinger** [comment]
Please try `Sleep 5`. 

If it doesn't work, try `SetOption127 1`

**Reply by Android-X13** [reply]
Hi, can you please give some more details?

Does that mean that the device is sleeping when I issue the web requests and that the nmap scan doesn't do anything at all (it just happens to coincidentally query the device when the device wakes up by itself)?

From the docs I understand that `Sleep 5` will make it sleep for 5 milliseconds instead of the default 50, but when? How often does a sleep cycle occur?
And if I use `SetOption127 1` will that be bad for the device?

PS. I didn't notice this was moved to Discussions so I created a second issue after the first one was auto-closed. I will close that one as well

**Reply by Jason2866** [reply]
sleep has nothing to do with miliseconds. And no `SetOption127 1` does no harm. It just disables wifi sleep.

**Reply by Android-X13** [reply]
I'm sorry but I don't follow... This is what I read in the docs regarding `Sleep`:
```
0 = turn sleep off
1..250 = set sleep duration in milliseconds to enable energy saving (default = 50)
```
So `Sleep 5` would mean "set sleep duration to 5 milliseconds instead of the default 50" (although that information means nothing if we don't know when each sleep cycle occurs).

**Reply by Jason2866** [reply]
A more precise description [https://tasmota.github.io/docs/Dynamic-Sleep/](https://tasmota.github.io/docs/Dynamic-Sleep/)

**Reply by s-hadinger** [reply]
Sleep 5 indeed makes a loop of roughly 5 ms between each iteration. In my experience it gives more chances for Wifi to detect incoming requests.
